File: Core/Entities/Question.cs
namespace Core.Entities;

public class Question : BaseEntity
{
    public string Title { get; set; }
    
    public string Text { get; set; } = null!;
    
    public string Slug { get; set; }
    
    public User? Author { get; set; }
    public int AuthorId { get; set; }
    
    public string? PictureUrl { get; set; }
    
    public DateTime DatePosted { get; set; } = DateTime.Now;
    
    public DateTime? LastModifiedDate { get; set; }

    public virtual List<Answer>? Answers { get; set; } = new List<Answer>();

    public virtual List<QuestionTag>? Tags { get; set; } = new List<QuestionTag>();

    public virtual List<Vote>? Votes { get; set; } = new List<Vote>();

    public int Score { get; set; } = 0;
}

File: StackOverflowFE/src/app/shared/models/question.ts
import { IAnswer } from './answer';
import { TagDto, TagReduced } from './tag';

export interface IQuestion {
    id: number;
    authorUsername: string;
    title: string;
    slug: string;
    text: string;
    creationDate: Date;
    lastModifiedDate: Date;
}

export interface IQuestionDetails {
    title: string;
    text: string;
    authorUsername: string;
    pictureUrl: string;
    datePosted: Date;
    lastModifiedDate: Date;
    answers: IAnswer[];
    tags: TagReduced[];
    score: number;
}

export interface IPostQuestion {
    id: number;
    title: string;
    text: string;
    pictureUrl: string;
    tags: TagDto[];
}

export interface IFilteredQuestions {
    questions: IQuestion[];
    totalNumberOfQuestions: number;
}
